examples/app_gradio.py

```diff
@@ -0,0 +1,133 @@
+            new_height = int((new_width / width) * height)
+            resized_image = image.resize((new_width, new_height), Image.LANCZOS)  # Replaced ANTIALIAS with LANCZOS
+        else:
+            # If width is already less than or equal to max_width, don't resize
+            resized_image = image
+
+        # Replace the image in the example
+        example[images_column] = resized_image
+        return example
+
+def run_vqasynth_pipeline(dataset: DatasetDict):
+
+    # Apply the resizing to your dataset
+    dataset = dataset.map(lambda example: resize_image_keep_aspect_ratio(example, images_column=images))
+
+    embedding_generator = EmbeddingGenerator()
+
+    dataset = dataset.map(lambda example: embedding_generator.apply_transform(example, images))
+
+    tag_filter = TagFilter()
+
+
+    # @title Set tags for filtering
+    include_tags = 'Park,Street,Home,Office,Garden,Playground,Warehouse,Shop,Library,Classroom,Kitchen,Living Room,Bedroom,Bathroom,Garage,Balcony,Alley,Sidewalk,Field,Meadow,Tennis court,Driveway,Lobby,Workshop,Gym' #@param {type:"string"}
+    exclude_tags = 'Aerial Photography,Single Subject,Macro Photography,Microscopic Views,Drawings,Unrealistic CGI,Graphs,Charts,Icons,Logos,Text Overlays,Minimalist,Art,Portraits,Close-up shot,Flat Art,Abstract Paintings,Silhouettes,Monochrome Photography,Stock Photos' #@param {type:"string"}
+
+    include_tags = include_tags.strip().split(",")
+    exclude_tags = exclude_tags.strip().split(",")
+
+    # Extract tags
+    dataset = dataset.map(lambda example: tag_filter.apply_transform(example, include_tags + exclude_tags))
+
+    # Filter by tags
+    dataset_filtered = dataset.filter(
+        lambda example: tag_filter.filter_by_tag(
+            example['tag'], include_tags, exclude_tags
+            )
+        )
+
+
+    depth = DepthEstimator()
+
+    dataset = dataset.map(lambda example: depth.apply_transform(example, images))
+
+    del embedding_generator
+    del tag_filter
+    del depth
+
+    localizer = Localizer()
+    dataset = dataset.map(lambda example: localizer.apply_transform(example, images))
+
+    spatial_scene_constructor = SpatialSceneConstructor()
+
+    # Storing pointclouds for viewing
+    point_cloud_dir = os.path.join(cache_dir, "pointclouds")
+    if not os.path.exists(point_cloud_dir):
+        os.makedirs(point_cloud_dir)
+
+    dataset = dataset.map(lambda example, idx: spatial_scene_constructor.apply_transform(example, idx, cache_dir, images), with_indices=True)
+    
+    pcds = dataset['train']['pointclouds'][0][0]
+
+    return pcds[1], 'ok'
+    
+
+
+def run(image: PIL.Image):
+    dataset = DatasetDict()
+    dataset["train"] = Dataset.from_dict({"images": [image]})
+
+    pcd_file, final_dataset = run_vqasynth_pipeline(dataset)
+    output_dir = "./vqasynth_output/gradio"
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    ply_file = f"{output_dir}/output.ply"
+    obj_file = f"{output_dir}/output.obj"
```

Comment:
Perhaps use a temp file/dir so that multiple requests don't overlap on the same naming convention?